https://issues.redhat.com/browse/ACM-20310 ClusterClaims doc update (#8115)
open-cluster-management/rhacm-docs: clusters/cluster_lifecycle/clusterclaims.adoc

If any of the previous claims are deleted or updated on managed cluster, they are restored or rolled back to a previous version automatically.

> After the managed cluster joins the hub cluster, the following two types of `ClusterClaim` created on the managedcluster are synchronized with the status of the `ManagedCluster` resource on the hub cluster. `ClusterClaims` with the `open-cluster-management.io/spoke-only` label are not synchronized.

Reviewer: ```suggestion
After the managed cluster joins the hub cluster, the following two types of `ClusterClaim` resources that are created on the managed cluster are synchronized with the status of the `ManagedCluster` resource on the hub cluster. `ClusterClaim` resources with the `open-cluster-management.io/spoke-only` label are not synchronized.
```

> After the managed cluster joins the hub cluster, the following two types of `ClusterClaim` created on the managedcluster are synchronized with the status of the `ManagedCluster` resource on the hub cluster. `ClusterClaims` with label `open-cluster-management.io/spoke-only` label are not synchronized:

Reviewer 2: is it should be: `ClusterClaims` with label `open-cluster-management.io/spoke-only` are not synchronized

Author: Updated

> `ClusterClaims` that {mce} manages:: Includes `id.k8s.io`, or labels with the `open-cluster-management.io` or `openshift.io` suffix. These `ClusterClaims` are synchronized to the `ManagedCluster` status on the hub cluster.

Reviewer: ```suggestion
`ClusterClaims` that {mce-short} manages:: Includes `id.k8s.io`, or labels with the `open-cluster-management.io` or `openshift.io` suffix. These `ClusterClaims` are synchronized to the `ManagedCluster` status on the hub cluster.
```

> `ClusterClaims` that {mce} manages:: Includes `id.k8s.io`, or labels with the `open-cluster-management.io` or `openshift.io` suffix. These `ClusterClaims are synchronized to the `ManagedCluster` status on the hub cluster.

Reviewer 2: changed to: These `ClusterClaims` are synchronized to the `ManagedCluster` status on the hub cluster.

Author: Updated

> `ClusterClaims` that you manage:: Without the `open-cluster-management.io` or `openshift.io` suffix. By default, up to 20 `ClusterClaims` are synchronized to the `ManagedCluster` status on the hub cluster.

Reviewer 3: ```suggestion
`ClusterClaims` that you manage:: Without the `open-cluster-management.io` or `openshift.io` suffix. By default, up to 20 `ClusterClaims` are synchronized to the `ManagedCluster` status on the hub cluster. Configure the `KlusterletConfig` API to change the number of `ClusterClaims` you manage that are synchronized. Complete the following steps:
```

For me it would be clearer to put the `configure the "KlusterletConfig" API...` and  `"ClusterClaims" that you manage::...` in the same paragraph. So users will know this configuration will not take effect on the MCE manages claims, what do you think?

Author: Updated

> To change the number of `ClusterClaims` that are synchronized, configure the `KlusterletConfig` API. Complete the following steps:

Reviewer 3: I am not sure if it is better to emphasize that this configuration only works for customer managing claims.

Author: I think it makes sense to mention it again, I changed the wording to emphasize it here.

> `ClusterClaims` that you manage:: Without the `open-cluster-management.io` or `openshift.io` suffix. By default, up to 20 `ClusterClaims` are synchronized to the `ManagedCluster` status on the hub cluster. Configure the `KlusterletConfig` API to change the number of `ClusterClaims` you manage that are synchronized. Complete the following steps:

Reviewer: While the code does have a "clusterclaims" (plural, written just like that) option in their `list`, the actual `kind` in the code is not plural. It is always `ClusterClaim`

I would recommend changing these instances to `ClusterClaim` resources. The title would change, as well.

Reviewer: When we code these, we are typically coding something that is in the YAML sample.

Reviewer 3: yes, usually we use lower case plural for a resource(clusterclaims), and use camel case singular for a kind(ClusterClaim)

Author: Updated!

. Create a `KlusterletConfig` and set `clusterClaimConfiguration.maxCustomClusterClaims` to your desired value. The maxmimum value is `100`. See the following example:

+
[source,yaml]
----
kind: KlusterletConfig
apiVersion: config.open-cluster-management.io/v1alpha1
metadata:
  name: klusterlet-cluster-claim-config
spec:
  clusterClaimConfiguration:
    maxCustomClusterClaims: 30
----

. Add an annotation to refer to the `KlusterletConfig` in the `ManagedCluster` resource. Run the following command:

+
[source,bash]
----
oc annotate managedcluster <cluster-name> agent.open-cluster-management.io/klusterlet-config="klusterlet-cluster-claim-config"
----
+
After adding the annotation, yor `ManagedCluster` resource resembles the following example:

> +

Reviewer: need a space here.
